kivakit-core/src/main/java/com/telenav/kivakit/core/thread/KivaKitThread.java

```diff
@@ -222,9 +222,7 @@ public static KivaKitThread run(Listener listener,
     private transient Runnable code;
```

Comment:
`code` should be final, and null when not needed.  Nothing reassigns it, and subclasses can't see it to reassign it.  Should not need `transient` - either 

 * You don't serialize it and nothing cares
 * You do serialize it and the runnable is present and serializable, or absent and null, and either way you're fine
 * You do serialize it and the runnable is present and not serializable - boom
   * As opposed to _you do serialize it and the runnable doesn't go with it, serialization succeeds and what reaches the other end of the wire doesn't work_ which is far more difficult to diagnose

Reply:
agree. fixed.

Reply:
Have a look. Should be resolved now.